awq/utils/calib_data.py

```diff
@@ -47,7 +47,7 @@ def get_calib_dataset(
             line = data[text_column]
             line = line.strip()
             line_encoded = tokenizer.encode(line)
-        if len(line_encoded) > 512:
+        if len(line_encoded) > max_seq_len:
```

Comment:
Should all calib input tokens be of same length (possibly left-padded up to the) `max_seq_len`?
Given that they're later concatenated along second dim ( [`cat_samples.shape`](https://github.com/casper-hansen/AutoAWQ/blob/af9242139b51dac5a61ca4c4cf9d93a443192eca/awq/utils/calib_data.py#L60)  will become `[1, N]`), and then sliced into tokens of sizes `[1, max_seq_len]`, which are later [batched](https://github.com/casper-hansen/AutoAWQ/blob/af9242139b51dac5a61ca4c4cf9d93a443192eca/awq/quantize/quantizer.py#L528).

For example if `max_seq_len=512`, `sample1`, `sample2` and `sample3` have shapes `[1, 400]`, `[1, 400]` and `[1, 300]` respectively, then the first row of the [batched samples](https://github.com/casper-hansen/AutoAWQ/blob/af9242139b51dac5a61ca4c4cf9d93a443192eca/awq/quantize/quantizer.py#L528) will be the original `sample1` + the first `112` tokens of `sample2`.
In this case we would have chopped off the first `112` tokens of `sample2` (which typically would contain the **system prompt**) making it a somewhat inaccurate sample.

Reply:
@attafosu it's a good point. However, to address this challenge would mean less efficient quantization or a much more complicated packing strategy as seen in axolotl. This could be a separate PR in itself to further optimize the packing of samples.

At the current point in time, I lean more towards doing evaluations to see if we can spot improvements in coding/long-context.

Reply:
I don't think padding each sentence independent  is necessary. In fact, in many LLMs (if I remember correctly, RoBERTa and GPT1.2.3 related papers), the training texts are often divided according to max_len for all datas (the problem you are worried about?).

Reply:
@WanBenLe My concern was more on losing the system prompts of certain samples in the calibration data when they're concatenated and sliced by max_len. For training due to the enormous amount of data, the effect may be quite negligble effect. Whereas for calibration/quantization one may have only fewer samples available which may not smoothen the potential effects - these are all theoretical, of course.

Reply:
> @attafosu it's a good point. However, to address this challenge would mean less efficient quantization or a much more complicated packing strategy as seen in axolotl. Essentially, figuring out which samples can go together within a given sequence length, e.g. `sample2 (200 tokens)` and `sample10 (310 tokens)` could fit together in a sequence of 512. This could be a separate PR in itself to further optimize the packing of samples.
> 
> At the current point in time, I lean more towards doing evaluations to see if we can spot improvements in coding/long-context.

Got it.